server/hooks.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+)
+
+func (p *Plugin) ChannelHasBeenCreated(c *plugin.Context, channel *model.Channel) {
+	// Set the description for any DM channels opened between Surveybot and a user
+	if channel.Type != model.CHANNEL_DIRECT {
+		return
+	}
+
+	if !strings.HasPrefix(channel.Name, p.botUserId+"__") && !strings.HasSuffix(channel.Name, "__"+p.botUserId) {
```

Comment:
This seems fine -- given the knowledge of how we encode direct channel names -- but would it worth querying for the users and explicitly checking for `p.botUserId` for clarity in the code?

Reply:
I don't think we need to do that. There's enough other code that relies on how these names are constructed that we'd require a lot of other changes if we were to change how those names are constructed.